### ISSUE TITLE
VS2019 (RC4) regression with certain use of in parameter semantics.

### ISSUE THREAD

**Issue by plynkus** [xamarin/xamarin-android#2906]
### Steps to Reproduce

1. Load VS2019 (RC4) and create an empty solution.
2. Create a blank Xamarin.Android application.
3. Create a blank Xamarin.iOS application.
4. To each, add the following snippet to the file for the main activity/app delegate:

```
    namespace Base
    {
        public abstract class Feature<T> where T : struct
        {
            public abstract void Update(in T value);
        }
    }

    public class Feature<T> : Base.Feature<T> where T : struct
    {
        public override void Update(in T value)
        {
            throw new System.NotImplementedException();
        }
    }

    struct Data
    {
        public float Value;
    }
```

5. To each the OnCreate(Bundle) and app delegate FinishedLaunching, simply instantiate the above with:

```
new Feature<Data>();
```

### Expected Behavior

A normal application launch.

### Actual Behavior

Crashes on each platform so long as the `in` parameter semantics are in use.  Removal of the `in` keywords from both `Feature<T>` signatures results in proper operation.

Exceptions on Android are of the form:

```
System.TypeLoadException: Could not find method '.ctor' due to a type load error:
Type Android.Feature`1[T] has invalid vtable method slot 4 with method none
```

...vs. corresponding outcomes on iOS:

```
System.ExecutionEngineException: Attempting to JIT compile method
'iOS.AppDelegate:FinishedLaunching (UIKit.UIApplication,Foundation.NSDictionary)'
while running in aot-only mode.
```

The identical snippets built (`in` parameters included) under the latest VS2017 (with appropriate compiler setting bumped up from defaults to allow `in` parameters at all) work as expected.  We tripped up on these during a VS2019 upgrade pass that included a couple use cases well boiled down to the snippets above.

There were no issues with the same semantics under VS2019 with UWP.

I have submitted this issue in the Android repo for starters, given that it seems likely the issue is of a common origin.  I am happy to issue a dup for the iOS repo if that turns out not to be the case.

**Comment by plynkus** [comment]
FYI: Also repros under VS2019 RTM.

**Comment by marek-safar** [comment]
This issue is fixed in Mono 2018-10 but fails with 2018-08. @lambdageek do you remember which change could fix it?

**Comment by pjcollins** [comment]
This issue should be resolved in the upcoming Visual Studio 2019 16.2 release. Public previews of this upcoming release are currently available.
